### PR TITLE
kou's jump model

### PR DESCRIPTION
**Pull request by rhandal-pfn**
- Addtion of Kou's jump model (see  A Jump-Diffusion Model for Option Pricing https://www.columbia.edu/~sk75/MagSci02.pdf)

### CODE REVIEW

**Comment by rhandal-pfn** [comment]
- In docstring examples large number of jumps for given parameters (2% up and 5% down) results in different jump based on python (and torch) version as random number generated with same seed differ. This occurs when number of jumps in a single day is quite large (accumulating up and down jumps with various sizes). Reducing jumps number makes it feasible for all versions to provide similar paths within tolerance.

File: pfhedge/stochastic/kou_jump.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    Examples:
+        >>> from pfhedge.stochastic import generate_kou_jump
+        >>>
+        >>> _ = torch.manual_seed(42)
+        >>> generate_kou_jump(2, 5, jump_per_year = 10.0)
+        tensor([[1.0000, 1.0021, 1.0055, 1.0089, 0.9952],
+                [1.0000, 1.0288, 1.0210, 1.0275, 1.0314]])
+    """
+    assert jump_eta_up > 1.0, "jump_eta_up must be larger than 1.0"
```
Comment (masanorihirano) [inline pfhedge/stochastic/kou_jump.py:102]: テストのコードではない場所ではassertは非推奨です。if文 + raise ValueErrorとかにしてください。

File: tests/instruments/primary/test_kou_jump.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import pytest
+import torch
+
+from pfhedge.instruments import KouJumpStock
+
+
+class TestKouJumpStock:
+    @pytest.mark.parametrize("seed", range(1))
+    def test_values_are_finite(self, seed, device: str = "cpu"):
```
Comment (masanorihirano) [inline tests/instruments/primary/test_kou_jump.py:9]: MartonJumpのテストとほぼ変わらないので、TestMartonJumpを少し加工して、classにslef.clsとかでclassを設定して、そことreprだけoverrideすればmartonjumpのテストを使いまわせるようにするといいと思いました。

**Review by masanorihirano — APPROVED** [review]

File: pfhedge/stochastic/kou_jump.py (hunk 1 above)
Comment (rhandal-pfn) [reply pfhedge/stochastic/kou_jump.py:102]: 了解です。直しておきます。

File: tests/instruments/primary/test_kou_jump.py (hunk 2 above)
Comment (rhandal-pfn) [reply tests/instruments/primary/test_kou_jump.py:9]: はい。やってみます。

**Comment by rhandal-pfn** [comment]
Changes
- common test for merton and kou's model
- the standard deviation for the jump test has to be increased as cpu test has no issues but gpu test has convergence issues (previous 0.4 was almost on the boundary of tolerance, ie. out of 100 tests with different random seeds a few would be fail for sure, with 0.5 std this number decreases: see also https://github.com/pfnet-research/pfhedge/pull/625).
